scripts/train/finetune_example/README.md

```diff
@@ -0,0 +1,144 @@
+     'response': 'Sunlight is the source of energy for nearly all ecosystems.'}
+```
+
+Now we have a local dataset and a preprocessing function that will map examples in our dataset to the desired prompt/response format. All that's left is to configure our training YAML to use them.
+
+## Our training YAML
+
+This is already taken care of in `gpt2-arc-easy.yaml`, specifically in the `train_loader` section that controls how the  train dataloader is built in the `train.py` script. (If you also have an accompanying validation dataset, you'd make similar changes in the `eval_loader` section). Let's take a closer look.
+
+<!--pytest.mark.skip-->
+```yaml
+train_loader:
+  name: finetuning
+  dataset:
+    ############
+    hf_name: json
+    hf_kwargs:
+      # Note: absolute paths for data_dir are more reliable;
+      # relative paths will be interpreted relative to whatever your
+      # working directory is when you run `train.py`
+      data_dir: finetune_example
+    # Note: `scripts/train` will be the working directory when resolving
+    # the preprocessing_fn import path
+    preprocessing_fn: finetune_example.preprocessing:multiple_choice
+    split: train
+```
+
+With this configuration, the finetuning code will *effectively* run the following code to construct the tokenized dataset used by the training dataloader:
+<!--pytest.mark.skip-->
+```python
+from datasets import load_dataset
+# (from "preprocessing_fn" in the config)
+from finetune_example.preprocessing import multiple_choice
+# (from "hf_name", "hf_kwargs", and "split" in the config)
+dataset = load_dataset('json', data_dir='finetune_example', split='train')
+
+def format_and_tokenize(example):
+   formatted_example = multiple_choice(example)
+   return tokenizer(text=formatted_example['prompt'], text_target=formatted_example['response'])
+
+tokenized_dataset = dataset.map(format_and_tokenize)
+```
+
+Hopefully this clarifies how the dataloader settings in the YAML affect dataset construction!
+
+As you can see, these config settings mainly just determine the inputs to `load_dataset`, from Hugging Face's `datasets` library. As such, we encourage you to review [their documentation](https://huggingface.co/docs/datasets/loading) to get a better sense of what our tooling will let you do!
+
+A couple notes on that:
+- You can use either the `data_dir` or `data_file` kwarg to tell `load_dataset` where your data live.
+- `data_dir` tends to be a bit more reliable, but requires that your actual data files within the directory are named `train`, `validation`, and/or `test`.
+- Whenever possible, use **absolute paths** instead of relative paths. Relative paths will depend on the working directory you are in when running the `train.py` script.
```

Comment:
```suggestion
- Whenever possible, use **absolute paths** instead of relative paths. Relative paths will depend on the working directory you are in when running the `train.py` script.

## ARC data

The data in `train.jsonl` is from the ARC corpus.

This work uses data from the AI2 Reasoning Challenge (ARC) 2018, created by and copyright [AI2 and Aristo​](https://allenai.org/data/arc)​. The data consists of 7,787 science exam questions and is intended for non-commercial, research purposes only. We use the Easy Dev split, consisting of 570 questions and answer, which we do not modify. The original data is available at https://allenai.org/data/arc and is licensed under a [Creative Commons Attribution-ShareAlike 4.0 International License](https://creativecommons.org/licenses/by-sa/4.0/).
```

Reply:
Thanks!

I trimmed `train.jsonl` to the first 100 examples to make the demonstration more lightweight. Do you have a suggestion for how to modify the above language to reflect that (if needed)?

Reply:
@samhavens I went ahead and updated the suggested change to reflect what I described. If it looks reasonable to you, please feel free to commit the suggestion.